### PR TITLE
First Turkish Retrieval dataset

### PR DESCRIPTION
**Pull request by asparius**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

It is an historical question answering dataset collected in Turkish. It is in the same format as NobQuad or Fquad. [Paper](https://ieeexplore.ieee.org/document/9559013) of the dataset

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

**Comment by asparius** [comment]
Did linting, but still fails.

File: mteb/tasks/Retrieval/tur/TurHistQuad.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                year={2021},
+                volume={},
+                number={},
+                pages={215-220},
+                keywords={Computer science;Computational modeling;Neural networks;Knowledge discovery;Information retrieval;Natural language processing;History;question answering;information retrieval;natural language understanding;deep learning;contextualized word embeddings},
+                doi={10.1109/UBMK52708.2021.9559013}}
+
+        """,
+        n_samples={"test": 1330, "train": 14221},
+        avg_character_length={"train": 1219.37, "test": 1513.83},
```
Comment (isaac-chung) [inline mteb/tasks/Retrieval/tur/TurHistQuad.py:43]:
```suggestion
        n_samples={"test": 1330},
        avg_character_length={"test": 1513.83},
```

File: mteb/tasks/Retrieval/tur/TurHistQuad.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            return
+
+        self.dataset = datasets.load_dataset(**self.metadata_dict["dataset"])
+
+        self.corpus = {}
+        self.relevant_docs = {}
+        self.queries = {}
+        text2id = {}
+
+        for split in self.dataset:
```
Comment (isaac-chung) [inline mteb/tasks/Retrieval/tur/TurHistQuad.py:63]: The evaluate method only consider `eval_splits`, so no need to go through this for the `train` split.

**Review by isaac-chung — COMMENTED** [review]
Great work! Just a few small comments. Feel free to add points as well.

File: docs/mmteb/points/533.jsonl [hunk 3]
```diff
@@ -0,0 +1,2 @@
+{"GitHub": "isaac-chung", "Review PR": 2}
+{"GitHub": "asparius", "New dataset": 2}
```
Comment (isaac-chung) [inline docs/mmteb/points/533.jsonl:2]:
```suggestion
{"GitHub": "asparius", "New dataset": 6}
```

**Review by isaac-chung — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**